profile/profile.go

```diff
@@ -674,6 +674,39 @@ func numLabelsToString(numLabels map[string][]int64, numUnits map[string][]strin
 	return strings.Join(ls, " ")
 }
 
+// SetLabel sets the specified key to the specified value for all samples in the
+// profile.
+func (p *Profile) SetLabel(key string, value []string) {
+	for _, sample := range p.Sample {
+		if sample.Label == nil {
```

Comment:
Could avoid lookup on the "no key" case as
```
if sample.Label == nil {
  sample.Label = map[string][]string{key: value}
} else {
  sample.Label[key] = value
}
```

Reply:
Done.